src/inspector/inspector_agent.cpp

```diff
@@ -393,40 +388,57 @@ void InspectorWrapConsoleCall(const v8::FunctionCallbackInfo<v8::Value> &args) {
                                 .ToLocalChecked());
 }
 
+class ServerCollection {
+ public:
+  std::shared_ptr<inspector::InspectorSocketServer> getServer(int port) {
+    const std::lock_guard<std::mutex> lock(mutex_server_init_);
+
+    std::shared_ptr<inspector::InspectorSocketServer> server;
+    // Reuse if the server is already running on the given port.
+    auto existingServers = servers_.find(port);
+    if (existingServers != servers_.end()) {
+      auto s = servers_[port].lock();
+      if (s)
+        server = std::move(s);
+      else
+        servers_.erase(port);
+    }
+
+    if (!server) {
+      auto delegate = std::make_unique<InspectorAgentDelegate>();
+      auto newserver =
+          std::make_shared<InspectorSocketServer>(std::move(delegate), port);
+      if (!newserver->Start()) {
+        std::abort();
+      }
 
-InspectorSocketServer& AgentImpl::ensureServer() {
-  if (server_) return *server_;
-
-  const std::lock_guard<std::mutex> lock(g_mutex_server_init_);
-  // Reuse if the server is already running on the given port.
-  auto existingServers = g_servers_.find(port_);
-  if (existingServers != g_servers_.end()) {
-    auto server = g_servers_[port_].lock();
-    if (server)
-      server_ = std::move(server);
-    else
-      g_servers_.erase(port_);
-  }
-
-  if (!server_) {
-    auto delegate = std::make_unique<InspectorAgentDelegate>();
-    auto newserver =
-        std::make_shared<InspectorSocketServer>(std::move(delegate), port_);
-    if (!newserver->Start()) {
-      std::abort();
+      server = std::move(newserver);
+      servers_[port] = server;
     }
-
-    server_ = std::move(newserver);
-    g_servers_[port_] = server_;
+
+    return server;
   }
-  
-  return *server_;
-}
+
+  void removeServer(int port) {
+    servers_.erase(port);
+  }
+
+  static ServerCollection& Instance() {
+    static ServerCollection instance;
+    return instance;
+  }
+
+ private:
+  std::mutex mutex_server_init_;
+  std::unordered_map<int, std::weak_ptr<inspector::InspectorSocketServer>> servers_;

```

Comment:
>int [](http://example.com/codeflow?start=21&length=3)

(nit) is this port? could you add comment like:
std::unordered_map<int / *port* /, std::we